### DISCUSSION TITLE
A problem about  circle filled with gradient transparency

### DISCUSSION THREAD

**Discussion by XianranZ** [openscenegraph/OpenSceneGraph#1278] [Q&A]
Hello, everyone!
I try to draw a circle filled by gradient color using OSG3.4.0. I want to achieve this effect via binding different gradient transparency for the points in different level. However, I don't get the result as expected. It seems that the gradient points are not the postion I set. 
**Below is my code.**
![微信图片_20231130093037](https://github.com/openscenegraph/OpenSceneGraph/assets/46214062/e27a7e44-9e90-4d83-b0f8-c6de92427cfd)
**And this is the running result**
![微信图片_20231130093955](https://github.com/openscenegraph/OpenSceneGraph/assets/46214062/ec3f34ae-7042-4e67-84df-32a46ba55587)
![微信图片_20231130094006](https://github.com/openscenegraph/OpenSceneGraph/assets/46214062/2d9319c2-e590-43ea-99c7-f7097162bb76)
I set the gradient point at the 70% of the circle radius. So I think the gradient shuold appear at that point. But the running result is not like this. 
Why?
If you know something about this, please give me some tips! thanks very much.

**Comment by robertosfield** [comment]
The whole thing seems a bit perverse, lots of verticals for something so simple, could you just not use a texture to provide the colour gradient?

As for what might be wrong, you don't say anything about how you intend the geometry data connected together as primitives, but I for sure a POLYGON won't be the answer - it maps to GL_POLYGON.  Please look up OpenGL docs on GL_POLYGON.

**Comment by FuXiii** [comment]
If you want the alpha channel to be applied, you'd better use `osg::BlendEquation` to apply `Color Blend` and `osg::StateSet::setRenderingHint(osg::StateSet::RenderingHint::TRANSPARENT_BIN)` to apply `transparent(queue)`.
